synapse/lib/lmdblayer.py

```diff
@@ -3,412 +3,284 @@
-            self.layr.spliced.set()
+        self.slab.forcecommit()
 
-    def getBuidProps(self, buid):
-        return self.buidcache.get(buid)
+        # wake any splice waiters and clear the splices out...
+        if self.splicelist:
+            self.spliced.set()
+            self.spliced.clear()
+            self.splicelist.clear()
 
-    def _getBuidProps(self, buid):
+    async def getBuidProps(self, buid):
 
         props = {}
 
-        if not self.buidcurs.set_range(buid):
-            return props
-
-        for lkey, lval in self.buidcurs.iternext():
-
-            if lkey[:32] != buid:
-                break
+        for lkey, lval in self.slab.scanByPref(buid, db=self.bybuid):
 
             prop = lkey[32:].decode('utf8')
             valu, indx = s_msgpack.un(lval)
             props[prop] = valu
 
         return props
 
-    def _storPropSet(self, oper):
+    async def _storPropSet(self, oper):
 
         _, (buid, form, prop, valu, indx, info) = oper
 
-        if len(indx) > 256: # max index size...
+        if len(indx) > MAX_INDEX_LEN:
             mesg = 'index bytes are too large'
             raise s_exc.BadIndxValu(mesg=mesg, prop=prop, valu=valu)
 
-        fenc = self.layr.encoder[form]
-        penc = self.layr.encoder[prop]
+        fenc = self.encoder[form]
+        penc = self.encoder[prop]
 
         univ = info.get('univ')
 
         # special case for setting primary property
         if not prop:
             prop = '*' + form
 
-        bpkey = buid + self.layr.utf8[prop]
+        bpkey = buid + self.utf8[prop]
 
-        cacheval = self.buidcache.cache.get(buid)
-        if cacheval is not None:
-            cacheval[prop] = valu
+        # FIXME:  might need to update any cortex buid cache
 
         bpval = s_msgpack.en((valu, indx))
 
         pvpref = fenc + penc
         pvvalu = s_msgpack.en((buid,))
 
-        byts = self.xact.replace(bpkey, bpval, db=self.layr.bybuid)
+        byts = self.slab.replace(bpkey, bpval, db=self.bybuid)
         if byts is not None:
 
             oldv, oldi = s_msgpack.un(byts)
 
-            self.xact.delete(pvpref + oldi, pvvalu, db=self.layr.byprop)
+            self.slab.delete(pvpref + oldi, pvvalu, db=self.byprop)
 
             if univ:
                 unkey = penc + oldi
-                self.xact.delete(unkey, pvvalu, db=self.layr.byuniv)
+                self.slab.delete(unkey, pvvalu, db=self.byuniv)
 
-        self.xact.put(pvpref + indx, pvvalu, dupdata=True, db=self.layr.byprop)
+        self.slab.put(pvpref + indx, pvvalu, dupdata=True, db=self.byprop)
 
         if univ:
-            self.xact.put(penc + indx, pvvalu, dupdata=True, db=self.layr.byuniv)
+            self.slab.put(penc + indx, pvvalu, dupdata=True, db=self.byuniv)
 
-    def _storPropDel(self, oper):
+    async def _storPropDel(self, oper):
 
         _, (buid, form, prop, info) = oper
 
-        self.buidcache.pop(buid)
+        # FIXME:  update any cortex-wide buid cache
```

Comment:
Do we still have the cortex wide cache issue?

Reply:
yes